compiler/test/dotty/tools/debug/DebugStepAssert.scala

```diff
@@ -0,0 +1,138 @@
+package dotty.tools.debug
+
+import com.sun.jdi.Location
+import dotty.tools.io.JPath
+import dotty.tools.readLines
+
+import scala.annotation.tailrec
+
+/**
+ * A debug step and an associated assertion to validate the step.
+ * A sequence of DebugStepAssert is parsed from the check file in tests/debug
+ */
+private[debug] case class DebugStepAssert[T](step: DebugStep[T], assert: T => Unit)(
+  using val location: CheckFileLocation
+)
+
+/** A location in the check file */
+private[debug] case class CheckFileLocation(checkFile: JPath, line: Int):
+  override def toString: String = s"$checkFile:$line"
+
+/** When a DebugStepAssert fails it throws a DebugStepException */
+private[debug] case class DebugStepException(message: String, location: CheckFileLocation) extends Exception
+
+private[debug] enum DebugStep[T]:
+  case Break(className: String, line: Int) extends DebugStep[Location]
+  case Step extends DebugStep[Location]
+  case Next extends DebugStep[Location]
+  case Eval(expression: String) extends DebugStep[Either[String, String]]
+
+private[debug] object DebugStepAssert:
+  import DebugStep.*
+  def parseCheckFile(checkFile: JPath): Seq[DebugStepAssert[?]] =
+    val sym = "[a-zA-Z0-9$.]+"
+    val line = "\\d+"
+    val trailing = s"\\s*(?:\\/\\/.*)?".r // empty or comment
```

Comment:
`/` is not a special character for regexes. There is no need to escape it. Also consider using `raw"` for regex strings:
```suggestion
    val trailing = raw"\s*(?://.*)?".r // empty or comment
```

Reply:
Are you sure you want to accept any whitespace? That's not consistent with the space that is required after the command names. Consider
```suggestion
    val trailing = raw" *(?://.*)?".r // empty or comment
```

Reply:
Yes I think it is convenient to accept trailing spaces. Otherwise parsing can fail and it is non-obvious why:
```
Parsing failed: step 4   
```

Reply:
done